### DISCUSSION TITLE
Networkk request after nextjs build

### DISCUSSION THREAD

**Discussion by mirik999** [vercel/next.js#15636] [Help]
Hello guys.
After nextjs build  it doesnt send any http requests to the api. 
* Api calls written in getStaticProps 

But in dev mode everything is okey.

Any suggestions ? i'm new in next.

**Comment by jamesmosier** [comment]
Are you using Next's API routes or another external API?

If you are using Next's API routes inside `getStaticProps` take a look at this section in the docs: https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly

**Comment by mirik999** [comment]
external API

**Comment by jamesmosier** [comment]
It's a little hard to say without seeing your code, but if you have an API call in `getStaticProps` then that API is called during `build` and not after. The reason in dev mode you are seeing it is because in dev mode `getStaticProps` is called on each page load for development purposes. In production this won't be called because it will have already been called when you built your application

**Reply by mirik999** [reply]
Yes , thats what happened , i cant see any updates on front-end. 
Thats my code 
```
export const getStaticProps: GetStaticProps = async (ctx) => {
  const response = await fetch(`${process.env.address}/api/posts/get`);
  const data = await response.json();

  return {
    props: {
      articles: data.posts,
    },
  };
};
```

**Reply by jamesmosier** [reply]
I guess I'm not understanding. 

This is the way it works:

1. You run `next build` and during that build `${process.env.address}/api/posts/get` is called
2. You run your app via `next start`
3. You should see **no** HTTP network requests for `${process.env.address}/api/posts/get` which is correct

**Reply by mirik999** [reply]
Yes.
